cmd/scollector/collectors/extrahop.go

```diff
@@ -0,0 +1,310 @@
+							metricsKept++
+						} else {
+							otherValues[dp.Timestamp] += dp.Value.(int64)
+							metricsDropped++
+						}
+					case "namedprotocols": //Only include protocols that have an actual name (SSL443 excepted)
+						if strings.Index(dp.Tags["proto"], "tcp") != 0 && strings.Index(dp.Tags["proto"], "udp") != 0 && (strings.Index(dp.Tags["proto"], "SSL") != 0 || dp.Tags["proto"] == "SSL443") { //The first characters are not tcp or udp.
+							*md = append(*md, dp)
+							metricsKept++
+						} else {
+							otherValues[dp.Timestamp] += dp.Value.(int64)
+							metricsDropped++
+						}
+					case "none": //Log everything. Is OK for viewing short timespans, but calculating, 2,000+ protocols over a multi-day window is bad for Bosun's performance
+						*md = append(*md, dp)
+						metricsKept++
+					}
+
+				}
+				//Take the consolidated values and add them now too
+				for k, v := range otherValues {
+					np := &opentsdb.DataPoint{
+						Metric:    xhMetricName,
+						Timestamp: k,
+						Tags:      opentsdb.TagSet{"vlan": fmt.Sprintf("%d", vlan.VlanId), "proto": extraHopOtherProtoName, "host": c.APIHost, "network": net.Name},
+						Value:     v,
+					}
+					*md = append(*md, np)
+				}
+				//And add the metadata for the metric
+				metadata.AddMeta(xhMetricName, nil, "rate", l7s.Rate, false)
+				metadata.AddMeta(xhMetricName, nil, "unit", l7s.Unit, false)
+				metadata.AddMeta(xhMetricName, nil, "desc", fmt.Sprintf("%s %s", l7s.Description, extraHopL7Description), false)
+			}
+		}
+	}
+
+	return nil
+}
+
+//These are used when looping through which L7 traffic to get. We want byte counts and packet counts, and this is the metadata that goes with them.
+var l7types = map[string]L7Stats{
+	"bytes": {Rate: metadata.Gauge, Unit: metadata.Bytes, Description: fmt.Sprintf("The number of bytes transmitted on this network.You can drill down by server, network, vlan and protocol for further investigations.")},
+	"pkts":  {Rate: metadata.Gauge, Unit: metadata.Counter, Description: fmt.Sprintf("The number of packets transmitted on this network. You can drill down by server, network, vlan and protocol for further investigations.")},
+}
+
+type L7Stats struct {
+	Rate        metadata.RateType
+	Unit        metadata.Unit
+	Description string
+}
+
+//Given the % value in the configuration file, calculate what the actual minimum value is for each of the time points returned by ExtraHop
+func calculateDataCutoff(k gohop.MetricResponseKeyed) map[int64]int64 {
+	sums := make(map[int64]int64)
+	rets := make(map[int64]int64)
+	for _, dp := range k.Stats {
+		for _, dv := range dp.Values {
+			for _, dw := range dv {
+				sums[dp.Time/1000] += dw.Value
+			}
+
+		}
+	}
+	for k, v := range sums {
+		rets[k] = int64(float64(v) * (1 - float64(extraHopTopProtoPerc)/100))
+		//log.Println(fmt.Sprintf("For timestamp %d: total of %d, to be in the top %d percent, must have over %d", k, v, extraHopTopProtoPerc, rets[k]))
+	}
+	return rets
+}
+
+/*
```

Comment:
Generally we don't merge in commented code. So this should probably be moved to another branch.